src/main/webapp/wise5/components/summary/summary-authoring/summary-authoring.component.ts

```diff
@@ -141,24 +169,34 @@ class SummaryAuthoringController extends EditComponentController {
     return false;
   }
 
-  showPromptFromOtherComponentChanged() {
+  showPromptFromOtherComponentChanged(): void {
     this.componentChanged();
   }
 
-  componentHasCorrectAnswer() {
+  componentHasCorrectAnswer(): boolean {
     const nodeId = this.authoringComponentContent.summaryNodeId;
     const componentId = this.authoringComponentContent.summaryComponentId;
     if (nodeId != null && componentId != null) {
       const component = this.ProjectService.getComponentByNodeIdAndComponentId(nodeId, componentId);
       if (component != null) {
-        const componentService = this.$injector.get(component.type + 'Service');
+        const componentService = this.getComponentService(component.type);
         return componentService.componentHasCorrectAnswer(component);
```

Comment:
Comment-1: Instead of re-defining getComponentService in this file, why don't we check for the specific component types (MC,OR,CM,Match) and use the existing getComponentService in ProjectService? This way, there's no need for extra parameters to this class's constructor nor the```initializeComponentTypeToService``` function

```suggestion
        if (['ConceptMap', 'Match', 'MultipleChoice', 'OpenResponse'].includes(component.type)) {
          const componentService = this.ProjectService.getComponentService(component.type);
          return componentService.componentHasCorrectAnswer(component);
        } else {
          return false;
        }
```